python/lsst/fgcmcal/fgcmBuildStarsTable.py

```diff
@@ -0,0 +1,347 @@
+# See COPYRIGHT file at the top of the source tree.
+#
+# This file is part of fgcmcal.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Build star observations for input to FGCM using sourceTable_visit.
+
+This task finds all the visits and sourceTable_visits in a repository (or a
+subset based on command line parameters) and extracts all the potential
+calibration stars for input into fgcm.  This task additionally uses fgcm to
+match star observations into unique stars, and performs as much cleaning of the
+input catalog as possible.
+"""
+
+import time
+
+import numpy as np
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.afw.table as afwTable
+
+from .fgcmBuildStarsBase import FgcmBuildStarsConfigBase, FgcmBuildStarsRunner, FgcmBuildStarsBaseTask
+from .utilities import computeApproxPixelAreaFields
+
+__all__ = ['FgcmBuildStarsTableConfig', 'FgcmBuildStarsTableTask']
+
+
+class FgcmBuildStarsTableConfig(FgcmBuildStarsConfigBase):
+    """Config for FgcmBuildStarsTableTask"""
+
+    referenceCCD = pexConfig.Field(
+        doc="Reference CCD for checking PSF and background",
+        dtype=int,
+        default=40,
+    )
+
+    def setDefaults(self):
+        self.instFluxField = 'ApFlux_12_0_instFlux'
+        self.localBackgroundFluxField = ''
+        self.apertureInnerInstFluxField = 'ApFlux_12_0_instFlux'
+        self.apertureOuterInstFluxField = 'ApFlux_17_0_instFlux'
+        self.psfCandidateName = 'Calib_psf_candidate'
```

Comment:
Because the transformation task which converts `src` to `sourceTable_visit` changes all the names to be more in line with the DPDD.  I've added a brief comment to this effect.